### DISCUSSION TITLE
Using tailwind 3/JIT within a component library causing build issues

### DISCUSSION THREAD

**Discussion by joecacti** [tailwindlabs/tailwindcss#8457] [Help]
I'm running into issues with the way tailwind 3 builds.  

For example I have created a component library vue/tailwind which contains the tailwind.config file and uses rollup to bundle this package up and use as an npm package.  

Pulling this package into our apps and then adding an unused class within the app doesn't recognize the class because it's not included in the npm package build.

Any suggestions how one might go about doing this so the watch/build inside the web app can add the newly used classes that aren't included in the node_modules ui-library tailwind pacakge?

Thanks

**Comment by hoshikitsunoda** [comment]
Is your component library set up to take in `className` prop?

```JavaScript
// /my-package/components/Button/index

const Button = ({
  className,
  children
})=> {
  const classes = `${className} p-4`;
  return (
    <button className={classes}>
      {children}
    </button>
  );
};
```

```JavaScript
// /consumer-app/components/CustomButton/index

import { Button } from "my-package"

const CustomButton = () => {
  return <Button className="text-red-500">Button!</Button>;
};
```

**Reply by hoshikitsunoda** [reply]
Another thing to check is if the package is included in `tailwind.config.js` in the consuming app.

```JavaScript
// /consumer-app/tailwind.config.js

module.exports = {
  mode: "jit",
  content: [
    "./src/**/*.{js,ts,jsx,tsx}",
    "./node_modules/my-package/dist/**/*.js",
  ],
};
```
